### DISCUSSION TITLE
User blocking general discussion

### DISCUSSION THREAD

**Discussion by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#621] [General]
This is a discussion about the blocking implementation you can leave your opinions and thoughts on this discussion as i think this is valuable feedback

**Comment by epicEaston197** [comment]
In my opinion the blocking feature needs a lot of work for example if somebody has me blocked I cannot see if they are speaking resulting in me talking over them and whenever they grab an item it disappears which could be confused for if someone is in a world layer or if there's avatar culling in the session

this can also heavily affect game worlds and other experiences if someone has you blocked like for example prop hunt you just won't be able to find the last person because you can't see them

**Reply by Frooxius** [reply]
The blocking feature definitely needs a lot more work and we plan on expanding and refining it significantly.

Our rationale for doing a quick version of it was that people are already getting harassed on the platform and the need to provide them with a tool to feel more comfortable outweighs the downsides and issues caused by the simpler implementation.

We understand it's not ideal situation, but it's "lesser of two evils" kind of thing.

**Comment by ShrikeAlvaron** [comment]
As awkward as it may be, not knowing that the person is there nor what they're doing is the entire point of blocking, if we start cutting out holes in it to let people know that a person is there and if they're talking or grabbing an object it starts defeating the entire purpose of blocking a person in the first place.

**Reply by Frooxius** [reply]
Our goal with expanding on the blocking system would be to provide a lot more fine grained controls. Part of that would be to determine the extent of the blocking you want to apply.

I think that's probably the best approach here. If given user wants to erase as much of given user's existence as possible, they will be able to do that.

If they want to know they're there and have some indication they're there and that they're talking, we'd want to provide that option too.

With blocking, these situations can be complex and differ case by case, so having better granularity is probably the best thing we can do.

**Reply by ShrikeAlvaron** [reply]
So long as options like those are determined by the blocker then that's fine, my concern was mainly around letting the blocked person get around parts of the block.  Giving the blocker the ability to better determine just how much of the other user is blocked (which could even include whether it's mutual perhaps) is fine, but for mutual blocks the blocked user should have to abide by the blocker's choices.

**Reply by Frooxius** [reply]
There will always be ways for determined blocked person to get around things. But at that point this will escalate things into severe harassment with the moderation team.

**Reply by epicEaston197** [reply]
I definitely want it to be world based so for prop hunt I can disable blocking so that the game can be played properly maybe put a warning on the session tab to tell people

**Reply by ShrikeAlvaron** [reply]
I don't agree with that, worlds should make it clear that they may not work properly if people have each other blocked in the session but nothing anyone can do in world should be able to override a safety/personal moderation feature like blocking.

Imagine that you block someone because they've harassed or even assaulted you, and you then join a world only to see them again because that world forced blocking off.  Or since it would be a setting that could be configured by at least the world host, imagine a host notices two people have each other blocked and disables blocking in that world.  There's all sorts of issues that could come of that, which is why safety features should not be able to be overridden so easily.

**Reply by win189** [reply]
i do feel Epiceaston has a point if a user is blocked or gets blocked it is only a visual for that person so for example if person a blocks person b and person a hosts a world somewhere later on person b is not blocked from joining that would which makes it that the host can't see  what person b is doing and might not notice that that person even came in the world as not everyone uses systems and checks the user list often to see if people joined in the world

I believe that if u have the person blocked it should auto-ban from hosted sessions other wise host cant be held reasonably responsible

I do believe the blocked part should just block the person from seeing any Cloud-spawned avatar and just be the default avatar  and nothing should be truly blocking voice  via this option 

i feel this option causes more issues than it can fix and would be better left to self-moderation if 2 people are having a problem they can just not be in the world with that person or grow up and be mature and Civil with each other if not host can just ban both users or one over the other depending on there opinions as it is there world

**Reply by Frooxius** [reply]
Our plan is to provide a method for game worlds and other sessions to override the blocking when it would affect the game. However on the flip side, users will also get a warning before joining this session, so they can make an informed decision if they're comfortable joining the world with the blocking suppressed or not.

**Comment by win189** [comment]
> There will always be ways for determined blocked person to get around things. But at that point this will escalate things into severe harassment with the moderation team.

Dear Sir u cannot effectively report who and what you cannot see as it would be hiersay

**Reply by Frooxius** [reply]
This is not true:
- Other people in the session can see the person and can provide testimony of their actions
- Even if you cannot see them, certain actions and events are still part of the logs. In fact, we used logs to determine culprits in a number of cases, when nobody witnessed them performing certain acts even without the blocking system
- You can see representations of that user - hierarchy, session list and so on
